packages/gatsby-plugin-netlify-cms/src/gatsby-node.js

```diff
@@ -33,6 +27,38 @@ function deepMap(obj, fn) {
   return obj
 }
 
+function replaceRule(value) {
+  // If `value` does not have a `test` property, it isn't a rule object.
+  if (!value || !value.test) {
+    return value
+  }
+
+  // when javascript we exclude node_modules
+  if (value.type === `javascript/auto` && value.exclude) {
+    return {
+      ...value,
+      exclude: new RegExp(
+        [value.exclude.source, `node_modules|bower_components`].join(`|`)
```

Comment:
We don't know for sure that the `exclude` is and will continue to be a RegExp, it could easily be an array. `castArray` is the safe route, its guaranteed not to fail, we always know what we're getting, and the output is equally compatible with Webpack.

Reply:
I'm checking that it's a regex now. If people have a custom js rule they should make sure to exclude node_modules themselves.